edi_storage_oca/migrations/16.0.1.0.0/pre-migrate.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+from odoo.tools.sql import create_column
+
+from odoo.addons.http_routing.models.ir_http import slugify
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    if not version:
+        return
+
+    required_modules = ("storage_backend", "edi_storage_oca")
+    if not env["ir.module.module"].search(
+        [("name", "in", required_modules), ("state", "=", "installed")]
+    ):
+        return
+    if not openupgrade.column_exists(env.cr, "edi_backend", "new_storage_id"):
```

Comment:
In order to avoid the create new column, change column to old and new to current you can do:
```python
        _column_renames = {
            "edi_backend": [
                ("storage_id", False),            ],
        }
        openupgrade.rename_columns(env.cr, _column_renames)
        create_column(env.cr, "edi_backend", "storage_id", "int4")
```
The old field will have the following name: `openupgrade.get_legacy_name("storage_id")`